src/internal/m365/collection/drive/collections_test.go

```diff
@@ -2962,46 +2963,96 @@ func (suite *OneDriveCollectionsUnitSuite) TestGet() {
-					data.NewState: {"folder", "file"},
+					data.NotMovedState: {"folder", "file"},
 				},
 				rootFolderPath1 + "/folder2": {
-					data.NewState: {"folder2", "file2"},
+					data.MovedState: {"folder2", "file2"},
 				},
 				rootFolderPath2: {
 					data.NewState: {"folder", "folder2"},
 				},
 				rootFolderPath2 + "/folder": {
-					data.NewState: {"folder", "file"},
+					data.NotMovedState: {"folder", "file"},
 				},
 				rootFolderPath2 + "/folder2": {
-					data.NewState: {"folder2", "file2"},
+					data.NotMovedState: {"folder2", "file2"},
 				},
 			},
 			expectedDeltaURLs: map[string]string{
 				driveID1: delta,
 				driveID2: delta2,
 			},
-			expectedFolderPaths: map[string]map[string]string{
+			expectedPreviousPaths: map[string]map[string]string{
 				driveID1: {
 					"root":    rootFolderPath1,
-					"folder":  rootFolderPath1 + "/folder",
+					"folder":  rootFolderPath1 + "/folder2", // note: this is a bug, but is currently expected
 					"folder2": rootFolderPath1 + "/folder2",
+					"folder3": rootFolderPath1 + "/folder2",
 				},
 				driveID2: {
 					"root":    rootFolderPath2,
 					"folder":  rootFolderPath2 + "/folder",
 					"folder2": rootFolderPath2 + "/folder2",
 				},
 			},
-			expectedDelList: pmMock.NewPrefixMap(map[string]map[string]struct{}{}),
-			doNotMergeItems: map[string]bool{
-				rootFolderPath1:              true,
-				rootFolderPath1 + "/folder":  true,
-				rootFolderPath1 + "/folder2": true,
-				rootFolderPath2:              true,
-				rootFolderPath2 + "/folder":  true,
-				rootFolderPath2 + "/folder2": true,
+			expectedDelList: pmMock.NewPrefixMap(map[string]map[string]struct{}{
+				rootFolderPath1: makeExcludeMap("file", "file2"),
+				rootFolderPath2: makeExcludeMap("file", "file2"),
+			}),
+			doNotMergeItems: map[string]bool{},
+		},
+		{
+			name:   "out of order item enumeration causes prev path collisions",
+			drives: []models.Driveable{drive1},
+			enumerator: mock.EnumerateItemsDeltaByDrive{
+				DrivePagers: map[string]*mock.DriveItemsDeltaPager{
+					driveID1: {
+						Pages: []mock.NextPage{{
+							Items: []models.DriveItemable{
+								driveRootItem("root"),
+								driveItem("fanny2", "fanny", driveBasePath1, "root", false, true, false),
+								driveItem("file2", "file2", driveBasePath1+"/fanny", "fanny2", true, false, false),
+								driveItem("nav", "nav", driveBasePath1, "root", false, true, false),
+								driveItem("file", "file", driveBasePath1+"/nav", "nav", true, false, false),
+							},
+						}},
+						DeltaUpdate: pagers.DeltaUpdate{URL: delta},
+					},
+				},
+			},
+			canUsePreviousBackup: true,
+			errCheck:             assert.NoError,
+			previousPaths: map[string]map[string]string{
+				driveID1: {
+					"root": rootFolderPath1,
+					"nav":  rootFolderPath1 + "/fanny",
+				},
+			},
+			expectedCollections: map[string]map[data.CollectionState][]string{
+				rootFolderPath1: {
+					data.NewState: {"fanny2"},
```

Comment:
I don't believe we should have folder info showing up in the parent of the folder. From my understanding, this is saying the metadata for this folder is going to be in root which is something we've been trying to avoid for a while now. It should only appear in the folder it refers to

Reply:
Hmm, I think this is a topic that's separate to the task at hand.  This behavior is consistent with all the other tests.  So either the tests are malformed in a consistent manner, or we have a bug.  Neither case really overlaps with this change.  If you think it deems further investigation, would you mind creating an issue ticket for it?